libarchive/archive_read_support_format_cab.c

```diff
@@ -2110,7 +2110,6 @@ lzx_decode_init(struct lzx_stream *strm, int w_bits)
 		ds->pos_tbl = malloc(sizeof(ds->pos_tbl[0]) * w_slot);
 		if (ds->pos_tbl == NULL)
 			return (ARCHIVE_FATAL);
-		lzx_huffman_free(&(ds->mt));
```

Comment:
This could introduce memory leaks on other code paths.  Would not it be better to fix it like this?
```diff
--- a/libarchive/archive_read_support_format_cab.c
+++ b/libarchive/archive_read_support_format_cab.c
@@ -3140,7 +3140,9 @@ static void
 lzx_huffman_free(struct huffman *hf)
 {
    free(hf->bitlen);
+   hf->bitlen = NULL;
    free(hf->tbl);
+   hf->tbl = NULL;
 }

 /*
```

Reply:
Can be added, but how does this resolve double free problem of hf->bitlen ?

Reply:
If you call `free(hf->bitlen)` and `hf->bitlen` is `NULL`, there is no double free there.

Reply:
in lzx_huffman_init() there is defined:

`if (hf->bitlen == NULL || hf->len_size != (int)len_size) {
	free(hf->bitlen);
	hf->bitlen = calloc(len_size,  sizeof(hf->bitlen[0]));
	if (hf->bitlen == NULL)
		return (ARCHIVE_FATAL);
	hf->len_size = (int)len_size;
} else
	memset(hf->bitlen, 0, len_size *  sizeof(hf->bitlen[0]));`

which means, it frees the bitlen

lzx_huffman_init() is called at line 2150, before that on line 2110 lzx_huffman_free() is called, so in the zx_huffman_init() function, bitlen is already freed and it is trying to free it again

I tested this with covscan and after applying this change, no other new errors or warnings were created, just the given important warning disapeared. Maybe I am missing something, but I do not see any problems here. Maybe upstream can give us some feedback.